src/components/data-management/UploadDetailsModal.jsx

```diff
@@ -1,58 +1,44 @@
 /* eslint-disable import/no-duplicates */
 import React, { useState, useEffect, useRef } from 'react';
-import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import dayjs from 'dayjs';
 import utc from 'dayjs/plugin/utc';
 import {
-  Modal, Button, Col, Row,
+  Modal, Button, Col, Row, Progress,
 } from 'antd';
 import { UploadOutlined } from '@ant-design/icons';
-import { useDispatch } from 'react-redux';
-import endUserMessages from 'utils/endUserMessages';
-import handleError from 'utils/http/handleError';
-import { createAndUploadSampleFile, fileObjectToFileRecord } from 'utils/upload/processUpload';
-
 import UploadStatus, { messageForStatus } from 'utils/upload/UploadStatus';
-import downloadSingleFile from 'utils/data-management/downloadSingleFile';
 
 dayjs.extend(utc);
 
 const UploadDetailsModal = (props) => {
-  const dispatch = useDispatch();
   const {
-    visible, onCancel, file,
+    visible, onCancel, file, extraFields, onUpload, onDownload, onRetry,
```

Comment:
any reason why we are passing `onUpload, onDownload, onRetry` as props from `SamplesTableCells` instead of defining them in here?

Reply:
Because this component is reused for the sample info and for the cell level meta info